File: backend/src/main/java/by/fpmibsu/findafriend/controller/commands/users/SetUserRoleHandler.java
package by.fpmibsu.findafriend.controller.commands.users;

import by.fpmibsu.findafriend.application.mediatr.RequestHandler;
import by.fpmibsu.findafriend.dataaccesslayer.user.UserDao;
import by.fpmibsu.findafriend.entity.User;

public class SetUserRoleHandler extends RequestHandler<Void, SetUserRoleCommand> {
    private final UserDao userDao;

    public SetUserRoleHandler(UserDao userDao) {
        this.userDao = userDao;
    }

    @Override
    public Void handle(SetUserRoleCommand request) throws Exception {
        userDao.updateRole(request.userId, User.Role.valueOf(request.newRole));
        return null;
    }
}

File: BackendNET/Backend/Commands/Users/SetUserRole/SetUserRoleHandler.cs
using Backend.Repository;
using MediatR;

namespace Backend.Commands.Users.SetUserRole;

public class SetUserRoleHandler : IRequestHandler<SetUserRoleCommand, Unit>
{
    private readonly IUserRepository _userRepository;
    private readonly ITokenRepository _tokenRepository;

    public SetUserRoleHandler(IUserRepository userRepository, ITokenRepository tokenRepository)
    {
        _userRepository = userRepository;
        _tokenRepository = tokenRepository;
    }

    public async Task<Unit> Handle(SetUserRoleCommand request, CancellationToken cancellationToken)
    {
        var user = await _userRepository.GetByIdAsync(request.UserId, cancellationToken);
        if (user is null)
        {
            return default;
        }

        user.Role = request.NewRole;
        await _userRepository.UpdateAsync(user, cancellationToken);
        await _tokenRepository.InvalidateTokensAsync(user.Id, cancellationToken);
        return default;
    }
}